src/lib/components/composites/paper-components/paper-view/PaperNavigationButton.svelte

```diff
@@ -2,47 +2,157 @@
+                }
+            });
+    }
+
+    async function getNextProjectPaper(paper: Project_Paper) {
+        await backendService
+            .getNextPaper({ id: paper.id })
+            .response.then((nextPaper) => {
+                nextProjectPaper = nextPaper;
+            })
+            .catch((error) => {
+                nextProjectPaper = undefined;
+                if (error.message !== "No%20next%20paper%20available.") {
+                    toast("Error, while loading the next paper:" + error);
+                }
+            });
+    }
+
+    async function getPreviousProjectPaper(paper: Project_Paper) {
+        await backendService
+            .getPreviousPaper({ id: paper.id })
+            .response.then((previousPaper) => {
+                previousProjectPaper = previousPaper;
+            })
+            .catch((error) => {
+                previousProjectPaper = undefined;
+                if (error.message !== "No%20previous%20paper%20available.") {
+                    toast("Error, while loading the next paper:" + error);
+                }
+            });
+    }
+
+    /**
+     * Pre-fetches the next and the previous paper to find out if the according button is enabled or
+     * disabled and the according paper does not have to be loaded on clicking of the button, but
+     * before.
+     */
+    $effect(() => {
+        // Loads the next or previous paper only if all necessary information is available.
+        if (loading) return;
+        (async () => {
+            const paper = await loadingProjectPaper;
+            if (!paper) return;
+            nextProjectPaper = previousProjectPaper = undefined;
+            buttonRightDisabled = buttonLeftDisabled = true;
+            if (direction === "right") {
+                if (reviewMode.isActivated) {
+                    await getNextProjectPaperToReview(paper);
+                } else {
+                    await getNextProjectPaper(paper);
+                }
+                buttonRightDisabled = nextProjectPaper === undefined;
+            } else if (direction === "left") {
+                if (reviewMode.isActivated) {
+                    previousProjectPaper = paperQueue[paperQueue.length - 1];
+                } else {
+                    await getPreviousProjectPaper(paper);
+                }
+                buttonLeftDisabled = previousProjectPaper === undefined;
+            }
+            showLoadingSpinner = false;
+        })();
+    });
+
+    /**
+     * Handles the navigation of the button with the direction "right". Therefore, it is checked whether the
+     * review mode is activated or not.
+     */
+    const navigate = async function () {
+        loading = showLoadingSpinner = true;
+        await navigatePaper(
+            direction,
+            loadingProjectPaper,
+            paperQueue,
+            loadingProject,
+            nextProjectPaper,
+            previousProjectPaper,
+        );
+    };
```

Comment:
Why not:

```suggestion
    async function navigate() {
        loading = showLoadingSpinner = true;
        await navigatePaper(
            direction,
            loadingProjectPaper,
            paperQueue,
            loadingProject,
            nextProjectPaper,
            previousProjectPaper,
        );
    };
```

And then below:

```
onclick={navigate}
```

Reply:
Good call. Thanks.